sdk/tests/utils/test-builders.ts

```diff
@@ -56,9 +56,16 @@ export const generateDefaultConfigParams = (
 
 export const createInOrderMints = async (context: WhirlpoolContext, reuseTokenA?: PublicKey) => {
   const provider = context.provider;
-  const tokenXMintPubKey = reuseTokenA || (await createMint(provider));
-  const tokenYMintPubKey = await createMint(provider);
-  return PoolUtil.orderMints(tokenXMintPubKey, tokenYMintPubKey).map(AddressUtil.toPubKey);
+  const tokenXMintPubKey = reuseTokenA ?? (await createMint(provider));
+
+  // ensure reuseTokenA is the first mint if reuseTokenA is provided
+  while (true) {
```

Comment:
this feels scary but i suppose it works XD. 

up to you since it is test code, but an alternative:
```
let ordered;
do {
    const tokenYMintPubKey = await createMint(provider);
   ordered = PoolUtil.orderMints(tokenXMintPubKey, tokenYMintPubKey).map(AddressUtil.toPubKey);
} while (!reuseTokenA || ordered[0].equals(reuseTokenA))

return ordered
```

Reply:
Thanks!
I've updated it to do-while style.